File: javascript/sum-of-multiples/sum-of-multiples.js
// REVIEW: I don't think writing a constructor this way is a good idea. Figure out.
function SumOfMultiples(input) {
	if (!(this instanceof SumOfMultiples)) {
	    if ('undefined' === typeof input) {
		    return new SumOfMultiples([3, 5]);
		} else {
		    return new SumOfMultiples(input);
		}
	}
	
	this.factors = input;
}

function multiple(number, factors) {
	var factorsLength = factors.length;
	for (var i = 0; i < factorsLength; ++i) {
		if (0 === (number % factors[i])) {
			return true;
		}
	}
	
	return false;
};

SumOfMultiples.prototype.to = function(limit) {
	var sum = 0;
	
	for (var i = 0; i < limit; ++i)	{
		if (multiple(i, this.factors)) {
			sum += i;
		}
	}
	
	return sum;
};

module.exports = SumOfMultiples;

File: python/sum-of-multiples/sum_of_multiples.py
def multiple(number, factors):
	factors_len = len(factors)
	for i in range(factors_len):
		if (0 == (number % factors[i])):
			return True
	
	return False

def sum_of_multiples(limit, factors=[3, 5]):
	# Remove zero as a factor since you can't divide things by zero
	if 0 in factors:
		del factors[0]
	
	s = 0
	
	for i in range(limit):
		if multiple(i, factors):
			s += i
	
	return s